scripts/extension-version.test.js

```diff
@@ -0,0 +1,149 @@
+const { expect } = require('chai')
+
+const { nextExtensionVersion } = require('./extension-version')
+
+describe('next extension version', () => {
+  it('it should work with stable channel and current version to new version schema', () => {
+    expect(
+      nextExtensionVersion({
+        prismaVersion: '2.0.0-beta.3',
+        extensionVersion: '0.0.35',
+      }),
+    ).to.eq('0.1.3')
+  })
+
+  it('it should work with stable channel and extension only push', () => {
+    expect(
+      nextExtensionVersion({
+        prismaVersion: '2.0.0-beta.3',
+        extensionVersion: '0.0.3',
+      }),
+    ).to.eq('0.1.3.1')
```

Comment:
Why does this return `0.1.3.1` while line 12 only returns `0.1.3`?

Reply:
Because the derived version matches the current extension version. So, this must be an "extension only" push.

Reply:
"derived version" does not match any of the words above. What does it mean?

Reply:
Derived version is the process of converting "2.0.0-beta.3" to "0.0.3". If this version matches the existing extension version, which in this case is "0.0.3". We append `.1`. 

I should actually write a test for going from `0.1.3.1` to `0.1.3.2`, doing that now.

Reply:
Added a test case for `0.1.3.1` - `0.1.3.2`

Reply:
So if the "derive" process returns the same version, you add `.1` at the end, if the current version already is that, you increment?